easybuild/easyblocks/generic/pythonpackage.py

```diff
@@ -449,6 +450,17 @@ def build_step(self):
                 env.setvar("CMAKE_INCLUDE_PATH", include_paths)
                 env.setvar("CMAKE_LIBRARY_PATH", library_paths)
 
+            curr_cc = os.getenv('CC')
+            python_ldshared = get_config_vars('LDSHARED')[0]
+            if python_ldshared:
+                if python_ldshared.split(' ')[0] != curr_cc:
+                    self.log.info("Python's value for $LDSHARED ('%s') doesn't use current $CC value ('%s'), fixing",
+                                  python_ldshared, curr_cc)
+                    env.setvar("LDSHARED", curr_cc + " -shared")
```

Comment:
@Micket Let's squeeze in another log message to get "full coverage" in the log w.r.t. `$LDSHARED` (and let's switch to positive logic while we're at it, easier to parse for humans):

```python
if python_ldshared:
    if python_ldshared.split(' ')[0] == curr_cc:
        self.log.info("Python's value of $LDSHARED ('%s') uses current $CC value ('%s'), so not touching it",
                      python_ldshared, curr_cc)
    else:
        ...
else:
    ...
```